internal/xds/translator/route.go

```diff
@@ -541,16 +545,15 @@ func buildXdsAddedHeaders(headersToAdd []ir.AddHeader) []*corev3.HeaderValueOpti
 				KeepEmptyValue: true,
 			})
 		} else {
-			for _, val := range header.Value {
```

Comment:
this will break the logic for requestHeaders

for responseHeaders, is the fix to avoid the split in the gateway-api layer https://github.com/envoyproxy/gateway/blob/32d61c492167a37311315ab04bed383fb6504b03/internal/gatewayapi/filters.go#L620 ?

please add a test case in the gateway-api and xds/translator layer